lisa/sut_orchestrator/libvirt/transformers.py

```diff
@@ -0,0 +1,523 @@
+            self._node.execute(
+                "cp .cargo/bin/* /usr/local/bin/",
+                shell=True,
+                sudo=True,
+                cwd=path,
+            )
+        self._node.execute("cargo --version", shell=True)
+
+    def install(self) -> str:
+        runbook: SourceInstallerSchema = self.runbook
+        self._log.info("Installing dependencies for Cloudhypervisor...")
+        self._install_dependencies()
+        self._log.info("Cloning source code of Cloudhypervisor ...")
+        code_path = _get_source_code(runbook, self._node, self.type_name(), self._log)
+        self._log.info("Building source code of Cloudhypervisor...")
+        self._build_and_install(code_path)
+        return _get_package_version(self._node, "cloud-hypervisor")
+
+
+class CloudHypervisorReleaseInstaller(CloudHypervisorInstaller):
+    @classmethod
+    def type_name(cls) -> str:
+        return "latest_release"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return BaseInstallerSchema
+
+    def install(self) -> str:
+        linux: Linux = cast(Linux, self._node.os)
+        linux.install_packages("jq")
+        command = (
+            "curl -s https://api.github.com/repos/cloud-hypervisor/"
+            "cloud-hypervisor/releases/latest | jq -r '.tag_name'"
+        )
+        latest_release_tag = self._node.execute(command, shell=True)
+        self._log.debug(f"latest tag: {latest_release_tag}")
+        wget = self._node.tools[Wget]
+        file_url = (
+            "https://github.com/cloud-hypervisor/cloud-hypervisor/"
+            f"releases/download/{latest_release_tag}/cloud-hypervisor"
+        )
+        file_path = wget.get(
+            url=file_url,
+            executable=True,
+            filename="cloud-hypervisor",
+        )
+        self._node.execute(f"cp {file_path} /usr/local/bin", sudo=True)
+        self._node.execute(
+            "chmod a+rx /usr/local/bin/cloud-hypervisor",
+            shell=True,
+            sudo=True,
+        )
+        self._node.execute(
+            "setcap cap_net_admin+ep /usr/local/bin/cloud-hypervisor",
+            sudo=True,
+        )
+        return _get_package_version(self._node, "cloud-hypervisor")
+
+
+def _get_source_code(
+    runbook: SourceInstallerSchema, node: Node, default_name: str, log: Logger
+) -> PurePath:
+    code_path = _get_code_path(runbook.path, node, default_name)
+
+    echo = node.tools[Echo]
+    echo_result = echo.run(str(code_path), shell=True)
+    code_path = node.get_pure_path(echo_result.stdout)
+
+    if not node.shell.exists(code_path):
+        # create and give permission on code folder
+        node.execute(f"mkdir -p {code_path}")
+        node.execute(f"chmod -R 777 {code_path}")
+
+    log.debug(f"cloning code from {runbook.repo} to {code_path}...")
+    git = node.tools[Git]
+    code_path = git.clone(url=runbook.repo, cwd=code_path)
+    git.fetch(cwd=code_path)
```

Comment:
It doesn't need to fetch, after clone, the code is latest.

Reply:
will remove.